Initial support for Angular expression syntax

deathaxe opened the pull request:
This PR...

resolves #30

1. adds syntax definitions for Angular expression syntax, which is a subset of JavaScript. see: https://angular.dev/guide/templates/expression-syntax

   Required rules are directly implemented in favor of re-using/including JavaScript to reduce dependencies and keep syntax lean (from parser's point of view).

2. adds syntax based folding rules to be able to fold control flow blocks regardless used indentation style.

3. adds symbol list definition to index `@let` variables.

deathaxe committed:
Initial support for Angular expression syntax

This commit...

1. adds syntax definitions for Angular expression syntax,
   which is a subset of JavaScript.
   see: https://angular.dev/guide/templates/expression-syntax

   Required rules are directly implemented in favor of re-using/including
   JavaScript to reduce dependencies and keep syntax lean (from parser's
   point of view).

2. adds syntax based folding rules to be able to fold control flow blocks
   regardless used indentation style.

3. adds symbol list definition to index `@let` variables.

princemaple commented:
This is looking great. Mostly working alreay. Only breakage I found is after a line like this:

`  @let currency = orders.value()?.[0]?.extra?.currency;`

The open angle bracket in `<table>` below this line shows

`text.html.ngx meta.let.value.ngx keyword.operator.comparison.ngx`

deathaxe committed:
Fix qualified member access

This commit adds contexts for JavaScript specific of qualified variables.

It uses branching to properly match all parts even across multiple lines.

deathaxe committed:
Assign ngx file extension

This commit assigns *.ngx file extensions as Angular Playground detects it
and assigns special syntax, too.

princemaple commented:
Perfection. Thank you so much!
